File: Domain/Item.cs
using System;

namespace Domain
{
    public class Item
    {
        public Guid Id { get; set; }

        public Guid DictionaryId { get; set; }
        public Dictionary Dictionary { get; set; }

        public string Original { get; set; }
        public string Translation { get; set; }
        public string Definition { get; set; }
        public string DefinitionOrigin { get; set; }
        public ItemType Type { get; set; }
        public DateTime CreationDate { get; set; }

        public bool IsStarred { get; set; }
        public bool IsLearned { get; set; }
        public int TotalRepeatsCount { get; set; }
        public int CorrectAnswersCount { get; set; }
        public bool GoesForNextDay { get; set; }
    }
}

File: client-app/src/app/models/item.ts
export enum ItemType {
	Word = 10,
	Phrase = 20,
}

export interface IItemsEnvelope {
	items: IItem[];
	queryResultSize: number;
}

export interface IItem {
	id: string;

	original: string;
	translation: string;
	definition: string | null;
	definitionOrigin: string | null;
	type: ItemType;
	creationDate: Date;

	isStarred: boolean;
	isLearned: boolean;

	totalRepeatsCount: number;
	correctAnswersCount: number;
	correctAnswersToCompletionCount: number;
}

export interface IEditItem {
	original: string;
	translation: string;
	definition: string | null;
	definitionOrigin: string | null;
}

export interface INewItem {
	original: string;
	translation: string;
	definition: string | null;
	definitionOrigin: string | null;
	type: ItemType;

	isStarred: boolean;
}
